app/src/main/java/com/tari/android/wallet/application/MigrationManager.kt

```diff
@@ -1,28 +1,47 @@
 package com.tari.android.wallet.application
 
-import androidx.lifecycle.viewModelScope
 import com.tari.android.wallet.BuildConfig
-import com.tari.android.wallet.ui.common.SimpleViewModel
+import com.tari.android.wallet.data.sharedPrefs.network.NetworkRepository
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+import java.io.File
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class MigrationManager @Inject constructor(private val manager: WalletManager) {
+class MigrationManager @Inject constructor(
+    private val walletManager: WalletManager,
+    private val application: TariWalletApplication,
+    private val networkRepository: NetworkRepository,
+) {
 
-    private val simpleViewModel = SimpleViewModel()
-
-    fun validateVersion(onValid: () -> Unit, onError: () -> Unit) {
+    suspend fun validateVersion(onValid: () -> Unit, onError: () -> Unit) {
         val walletVersion = getCurrentWalletVersion()
 
         if (walletVersion.isEmpty() || DefaultArtifactVersion(walletVersion) < DefaultArtifactVersion(BuildConfig.LIB_WALLET_MIN_VALID_VERSION)) {
-            simpleViewModel.viewModelScope.launch(Dispatchers.Main) { onError() }
+            onError()
         } else {
-            simpleViewModel.viewModelScope.launch(Dispatchers.Main) { onValid() }
+            onValid()
+        }
+
+        if (walletVersion.isNotEmpty() && DefaultArtifactVersion(walletVersion) >= DefaultArtifactVersion(PEER_DB_MIGRATION_MIN_VERSION)) {
```

Comment:
Consider changing `>=` to `<`. This logic execute `performPeerDbMigration()` only when wallet version is equal or greater than the rc.8. But this migration should happen **only once** when db is moving to rc.8 from the any previous version.